test/e2e/network/kube_proxy.go

```diff
@@ -184,42 +225,52 @@ var _ = SIGDescribe("Network", func() {
 		<-time.After(time.Duration(1) * time.Second)
 
 		ginkgo.By("Checking /proc/net/nf_conntrack for the timeout")
-		// If test flakes occur here, then this check should be performed
-		// in a loop as there may be a race with the client connecting.
-		e2essh.IssueSSHCommandWithResult(
-			fmt.Sprintf("sudo cat /proc/net/nf_conntrack | grep 'dport=%v'",
-				testDaemonTCPPort),
-			framework.TestContext.Provider,
-			clientNodeInfo.node)
-
-		// Timeout in seconds is available as the fifth column from
-		// /proc/net/nf_conntrack.
-		result, err := e2essh.IssueSSHCommandWithResult(
-			fmt.Sprintf(
-				"sudo cat /proc/net/nf_conntrack "+
-					"| grep 'CLOSE_WAIT.*dst=%v.*dport=%v' "+
-					"| tail -n 1"+
-					"| awk '{print $5}' ",
-				serverNodeInfo.nodeIP,
-				testDaemonTCPPort),
-			framework.TestContext.Provider,
-			clientNodeInfo.node)
-		framework.ExpectNoError(err)
-
-		timeoutSeconds, err := strconv.Atoi(strings.TrimSpace(result.Stdout))
-		framework.ExpectNoError(err)
-
 		// These must be synchronized from the default values set in
 		// pkg/apis/../defaults.go ConntrackTCPCloseWaitTimeout. The
 		// current defaults are hidden in the initialization code.
 		const epsilonSeconds = 60
 		const expectedTimeoutSeconds = 60 * 60
-
-		framework.Logf("conntrack entry timeout was: %v, expected: %v",
-			timeoutSeconds, expectedTimeoutSeconds)
-
-		gomega.Expect(math.Abs(float64(timeoutSeconds - expectedTimeoutSeconds))).Should(
-			gomega.BeNumerically("<", (epsilonSeconds)))
+		// Timeout in seconds is available as the fifth column from
+		// /proc/net/nf_conntrack.
+		re := regexp.MustCompile(
+			fmt.Sprintf(
+				`^(?:\w+\s+){4}(\d+)\s+CLOSE_WAIT.*dst=%s.*dport=%d`,
+				serverNodeInfo.nodeIP,
+				testDaemonTCPPort))
+		// Obtain the corresponding conntrack entry on the node
+		// the nf_conntrack file is checked on the pod client
+		// and it's mounted on /rootfs/proc/net
+		// retry in a loop to avoid flakes
+		cmd = fmt.Sprintf("cat /rootfs/proc/net/nf_conntrack "+
+			"| grep 'CLOSE_WAIT.*dst=%v.*dport=%v' "+
+			"| tail -n 1",
+			serverNodeInfo.nodeIP,
+			testDaemonTCPPort)
+		if err := wait.PollImmediate(5*time.Second, 30*time.Second, func() (bool, error) {
+			result, err := framework.RunHostCmd(fr.Namespace.Name, "e2e-net-exec", cmd)
+			// retry if we can't obtain the conntrack entry
+			if err != nil {
+				framework.Logf("failed to obtain conntrack entry: %v %v", result, err)
+				return false, nil
+			}
+			match := re.FindStringSubmatch(result)
```

Comment:
Would this be simpler if we just took the result line, broke it with `strings.Fields`, and then looked at `[4]` ?

I love regex, but it requires brainpower, which is a bad thing in a test